trustfall_proc_macros/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+use quote::quote;
+
+#[proc_macro_derive(VariantsAsVertexTypes)]
```

Comment:
Suggestions for trait name: `TrustfallEnumVertex`? Describes what it is, `Trustfall` makes sure of name conflicts, `EnumVertex` makes it clear that yes it is a special type of `Vertex` made of an enum?

Reply:
Love it!